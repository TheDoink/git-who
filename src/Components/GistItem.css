#gistTitle {
	text-align: center;
	font-size:3vw;
	margin-top:10px;
}