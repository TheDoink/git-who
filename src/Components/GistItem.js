import React, { Component } from 'react';
import './GistItem.css';
import GistFileItem from './GistFileItem';

import $ from 'jquery';

class GistItem extends Component {

  constructor(props) {
    super(props);
    this.state = {
      gist: {         // The gist object
        files: []
      },          
      loading: false,  // Whether or not we're in the process of getting more
      failure: false,
    };
  };

  // Getting the contents of the gist
  getGist(gistId) {

    // Ensure we actually have a good user
    if(gistId) {
      this.setState({loading: true}, function() {
        // Try and get that username...
        $.get(`https://api.github.com/gists/${gistId}`,

          // If we get a good response...
          (success, text, obj) => {

            // Convert the files object to an array for easier iterating
            success.files = Object.keys(success.files).map((obj) => {
              let ret = success.files[obj];
              ret.content.replace(" ", "&nbsp;");
              ret.content = ret.content.split("\n");
              return ret;
            });

            // Set the contents
            this.setState({gist: success, loading:false, failure: false}, function() {});

          })

          // If we get anything else...
          .fail((failure, text, error) => {
            this.setState({failure: true, loading:false});
          });
      });
    }
  }

  /* Only get repos if we have a new username, and if it's good */
  componentWillReceiveProps(nextProps) {
    if(nextProps.gistId) {

      this.getGist(nextProps.gistId);
    }

    // Even if no good gist, reset the data
    this.setState({gist: {files: []}}, function() {});
  };

  componentWillMount() {
    this.getGist(this.props.gistId);
  }

  // Quick function to error check and see who owns the gist
  getOwner() {
    let ret = "";
    if(this.state.gist && this.state.gist.owner) {
      ret = this.state.gist.owner.login
    }
    
    return ret;
  }

  render() {
    
    return (

        // {/* Always hide while loading */}
        <div className={this.state.loading ? 'hidden' : ''}>

          {/* Always hide if there was an error */}
          <div className={this.state.failure ? 'hidden' : ''}>

            {/* Gist title info */}
            <div id="gistTitle">
              <div>Gist: {this.state.gist.id}</div>
              <div>By: {this.getOwner.bind(this)()}</div>
            </div>

            {/* Listing out the files */}
            <div>
              {this.state.gist.files.map(item => (
                <GistFileItem key={item.filename} file={item} numLines={item.content.length.toString().length}>
                  
                </GistFileItem>
                
              ))}
            </div>
          </div>

          {/*Error message if there's a problem */}
          <div className={!this.state.failure ? 'hidden' : ''}>
            <div className="failText">
              A problem occured getting this gist!
            </div>
          </div>
        </div>
    );
  }
}

export default GistItem;
