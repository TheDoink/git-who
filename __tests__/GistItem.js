// var NameLink = require('../src/Components/NameLink');
import React from 'react';
import renderer from 'react-test-renderer';
import GistItem from '../src/Components/GistItem';

it('lines up with the snapshot', () => {
  let testMe = renderer.create(<GistItem gistId="d2eace7877e2aeaf942de0f8b3e7275c"></GistItem>).toJSON();

  expect(testMe).toMatchSnapshot();
  // testMe.getGist('d2eace7977e2aeaf942de0f8b3e7275c');

  
});
