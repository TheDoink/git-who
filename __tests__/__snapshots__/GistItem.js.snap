// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lines up with the snapshot 1`] = `
<div
  className="hidden"
>
  <div
    className=""
  >
    <div
      id="gistTitle"
    >
      <div>
        Gist: 
      </div>
      <div>
        By: 
        <span>
          <a
            href="#"
            onClick={[Function]}
          >
            
          </a>
        </span>
      </div>
    </div>
    <div />
  </div>
  <div
    className="hidden"
  >
    <div
      className="failText"
    >
      A problem occured getting this gist!
    </div>
  </div>
</div>
`;
